### ISSUE TITLE
Server crash in 1.13.1 due to "Writing into PalettedContainer from multiple threads"

### ISSUE THREAD

**Issue by basicmark** [boy0001/FastAsyncWorldedit#1144]
**Debug paste link**:
N/A, server crash

**Description of the problem:**
When using brushes sometimes the server crashes.

Summary
`java.lang.IllegalStateException
        at net.minecraft.server.v1_13_R2.DataPaletteBlock.b(SourceFile:43)
        at net.minecraft.server.v1_13_R2.DataPaletteBlock.setBlock(SourceFile:110)
        at net.minecraft.server.v1_13_R2.ChunkSection.setType(SourceFile:68)
        at net.minecraft.server.v1_13_R2.Chunk.a(Chunk.java:488)
        at net.minecraft.server.v1_13_R2.World.setTypeAndData(World.java:318)
        at net.minecraft.server.v1_13_R2.FluidTypeFlowing.a(FluidTypeFlowing.java:491)
        at net.minecraft.server.v1_13_R2.Fluid.a(SourceFile:56)
        at net.minecraft.server.v1_13_R2.WorldServer.a(WorldServer.java:668)
        at net.minecraft.server.v1_13_R2.TickListServer.a(TickListServer.java:77)
        at net.minecraft.server.v1_13_R2.WorldServer.q(WorldServer.java:660)
        at net.minecraft.server.v1_13_R2.WorldServer.doTick(WorldServer.java:301)
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:952)
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:411)
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831)
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729)
        at java.lang.Thread.run(Thread.java:748)
`

Full crash report:
https://hastebin.com/ayoxunivuq.pl

**How to replicate**:
Use blendball brush, it's not 100% repoduceable and from our side not clear what difference there is between when it works correctly and when it doesn't. Could be chunk boundary issue or maybe timing issue? 

**Checklist**:

- [N/A] I included a `/fawe debugpaste` link
- [X] I made sure there aren't duplicates of this report [(Use Search)](https://github.com/boy0001/FastAsyncWorldedit/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [X] I made sure I am using an up-to-date version of FAWE
- [X] I Made sure the bug/error is not caused by any other plugin

**Comment by Miclebrick** [comment]
It's a thread safety issue, so it won't be consistently reproducible. If you place blocks in the area at the same time that'll make it more likely to happen. I think the 1.13 version is still very WIP though.

**Comment by CommanderTvis** [comment]
This appears for me too. 

```
java.lang.IllegalStateException
	at net.minecraft.server.v1_13_R2.DataPaletteBlock.b(SourceFile:43)
	at net.minecraft.server.v1_13_R2.DataPaletteBlock.setBlock(SourceFile:110)
	at net.minecraft.server.v1_13_R2.ChunkSection.setType(SourceFile:68)
	at net.minecraft.server.v1_13_R2.Chunk.setType(Chunk.java:493)
	at net.minecraft.server.v1_13_R2.World.setTypeAndData(World.java:313)
	at net.minecraft.server.v1_13_R2.BlockCactus.a(BlockCactus.java:42)
	at net.minecraft.server.v1_13_R2.Block.b(Block.java:360)
	at net.minecraft.server.v1_13_R2.IBlockData.b(SourceFile:265)
	at net.minecraft.server.v1_13_R2.WorldServer.n_(WorldServer.java:537)
	at net.minecraft.server.v1_13_R2.WorldServer.doTick(WorldServer.java:287)
	at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:952)
	at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417)
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831)
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729)
	at java.lang.Thread.run(Unknown Source)
```

I'm using latest stable build on 1.13.2. Game crashes on //copy call.

**Comment by CommanderTvis** [comment]
I think that it's FAWE bug, because vanilla WorldEdit is valid

**Comment by Pugabyte** [comment]
Also experiencing this issue. 
![image](https://user-images.githubusercontent.com/12417961/52744867-c0ad3280-2fab-11e9-9547-89f6f66f9470.png)

![image](https://user-images.githubusercontent.com/12417961/52744991-0f5acc80-2fac-11e9-94d6-ed54de56d2d4.png)

**Comment by CommanderTvis** [comment]
My current workaround for this problem is just removing all the tile-entities (chest, furnaces, redstone). It helps :\

**Comment by N0tMyFaultOG** [comment]
Your ticket has been closed because it falls into one of the following categories:

- Your issue has been fixed in a newer version of FastAsyncWorldEdit for versions 1.8 - 1.12.2, get the latest build [here](https://ci.athion.net/job/FastAsyncWorldEdit/lastSuccessfulBuild/artifact/target/) matching the platform you need.

- Your issue targets minecraft version 1.13 or higher. This is the wrong repository then. Move to [this repository](https://github.com/IntellectualSites/FastAsyncWorldEdit) and open a bug report here.
BUT BEWARE: You should consider updating to 1.15.2, using the latest Fawe build for 1.15.2 (Build 255 when writing this, get it [here](https://ci.athion.net/job/FastAsyncWorldEdit-1.15/)) and reproducing the issue in a clean environment (fresh setup Spigot or Paper server).

- Your issue has been fixed in a build of FAWE for 1.15 or an issue exists on our new repository.

- You either didn't fill out the template or deleted it.

If none of the bullet points suits, head to the second one.
